### PR TITLE
Add dbt project

### PR DESCRIPTION
**Pull request by H1mansh0**

### CODE REVIEW

**Comment by maksymloktionov** [comment]
Can you make it the way that all dbt related content is under `_dbt` directory? We should have
```
_dbt/models
_dbt/tests
...
```

File: _dbt/schema_imgs/OLAP.png
Comment (maksymloktionov) [inline _dbt/schema_imgs/OLAP.png:1]: Could you replace this with the dag generated by dbt docs generate command?

**Comment by maksymloktionov** [comment]
What is the `dbt_support` folder? I believe you can put everything "dbt-related" under `_dbt` folder

File: _dbt/dbt_support/dbt/Dockerfile.dbt
Comment (maksymloktionov) [inline _dbt/dbt_support/dbt/Dockerfile.dbt:1]: You can move both dockerfile and python requirements to `_dbt`, it should be easier to spinup everything from root

File: _dbt/dbt_project.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+# Configuring models
+# Full documentation: https://docs.getdbt.com/docs/configuring-models
+
+# In this example config, we tell dbt to build all models in the example/
+# directory as views. These settings can be overridden in the individual model
+# files using the `{{ config(...) }}` macro.
+models:
+  football_dbt:
+    staging:
```
Comment (maksymloktionov) [inline _dbt/dbt_project.yml:32]: Could you play around with different materializations? e.g. I want `staging` models to be incrementals

File: _dbt/dbt_support/parser/Dockerfile.python
Comment (maksymloktionov) [inline _dbt/dbt_support/parser/Dockerfile.python:1]: I think you can merge both _dbt and python parsing environments into 1. I mean you can have 1 dockerfile and 1 requirements file

File: _dbt/.gitignore
Comment (maksymloktionov) [inline _dbt/.gitignore:1]: Could you add .user.yml file here as well?

**Comment by maksymloktionov** [comment]
Could you add tests to dimensional models? Also, it would be great to name` .yml` files accordingly, e.g. if you are adding tests/metadata for `fact_some_model` dbt model, then the yml file is better to name as `fact_some_model.yml`

**Comment by maksymloktionov** [comment]
Please move `starter-dbt.sh` to the _dbt directory. I see that some models are still without tests(please double check again). Also, `starter-dbt-.sh` does not work for me at the moment
```
maksym@Techstacks-Laptop vlad-de-internship % bash starter-dbt.sh
[+] Building 3.6s (7/8)                                                                                                                docker-container:mybuilder
 => [pars_to_db internal] load build definition from Dockerfile                                                                                              0.0s
 => => transferring dockerfile: 642B                                                                                                                         0.0s
 => [pars_to_db internal] load metadata for docker.io/library/python:3.11.4                                                                                  0.5s
 => [pars_to_db internal] load .dockerignore                                                                                                                 0.0s
 => => transferring context: 2B                                                                                                                              0.0s
 => [pars_to_db internal] load build context                                                                                                                 0.0s
 => => transferring context: 3.69kB                                                                                                                          0.0s
 => [pars_to_db 1/4] FROM docker.io/library/python:3.11.4@sha256:85b3d192dddbc96588b719e86991e472b390805a754681a38132de1977d8e429                            0.0s
 => => resolve docker.io/library/python:3.11.4@sha256:85b3d192dddbc96588b719e86991e472b390805a754681a38132de1977d8e429                                       0.0s
 => CACHED [pars_to_db 2/4] COPY . /app                                                                                                                      0.0s
 => ERROR [pars_to_db 3/4] RUN apt-get update && apt-get install -y     curl     unzip     && rm -rf /var/lib/apt/lists/*     && mkdir app/data     && mkdi  3.0s
------
 > [pars_to_db 3/4] RUN apt-get update && apt-get install -y     curl     unzip     && rm -rf /var/lib/apt/lists/*     && mkdir app/data     && mkdir -p /root/.kaggle/     && mv app/kaggle.json /root/.kaggle/kaggle.json     && chmod 600 /root/.kaggle/kaggle.json     && pip3 install -r app/requirements.txt     && kaggle datasets download -d martj42/international-football-results-from-1872-to-2017     && unzip *.zip -d ./app/data     && rm -rf *.zip     && cd app:
0.158 Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
0.297 Get:2 http://deb.debian.org/debian bookworm-updates InRelease [55.4 kB]
0.349 Get:3 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
0.399 Get:4 http://deb.debian.org/debian bookworm/main arm64 Packages [8688 kB]
0.723 Get:5 http://deb.debian.org/debian bookworm-updates/main arm64 Packages [2468 B]
0.771 Get:6 http://deb.debian.org/debian-security bookworm-security/main arm64 Packages [203 kB]
1.438 Fetched 9147 kB in 1s (6643 kB/s)
1.438 Reading package lists...
1.739 Reading package lists...
2.079 Building dependency tree...
2.152 Reading state information...
2.236 unzip is already the newest version (6.0-28).
2.236 The following additional packages will be installed:
2.236   libcurl3-gnutls libcurl4 libcurl4-openssl-dev
2.237 Suggested packages:
2.237   libcurl4-doc libidn-dev libldap2-dev librtmp-dev libssh2-1-dev
2.260 The following packages will be upgraded:
2.260   curl libcurl3-gnutls libcurl4 libcurl4-openssl-dev
2.357 4 upgraded, 0 newly installed, 0 to remove and 117 not upgraded.
2.357 Need to get 1510 kB of archives.
2.357 After this operation, 46.1 kB disk space will be freed.
2.357 Get:1 http://deb.debian.org/debian bookworm/main arm64 libcurl4-openssl-dev arm64 7.88.1-10+deb12u8 [475 kB]
2.554 Get:2 http://deb.debian.org/debian bookworm/main arm64 curl arm64 7.88.1-10+deb12u8 [309 kB]
2.563 Get:3 http://deb.debian.org/debian bookworm/main arm64 libcurl4 arm64 7.88.1-10+deb12u8 [365 kB]
2.602 Get:4 http://deb.debian.org/debian bookworm/main arm64 libcurl3-gnutls arm64 7.88.1-10+deb12u8 [360 kB]
2.734 debconf: delaying package configuration, since apt-utils is not installed
2.751 Fetched 1510 kB in 0s (4378 kB/s)
(Reading database ... 23997 files and directories currently installed.)
2.775 Preparing to unpack .../libcurl4-openssl-dev_7.88.1-10+deb12u8_arm64.deb ...
2.778 Unpacking libcurl4-openssl-dev:arm64 (7.88.1-10+deb12u8) over (7.88.1-10+deb12u1) ...
2.832 Preparing to unpack .../curl_7.88.1-10+deb12u8_arm64.deb ...
2.835 Unpacking curl (7.88.1-10+deb12u8) over (7.88.1-10+deb12u1) ...
2.867 Preparing to unpack .../libcurl4_7.88.1-10+deb12u8_arm64.deb ...
2.873 Unpacking libcurl4:arm64 (7.88.1-10+deb12u8) over (7.88.1-10+deb12u1) ...
2.919 Preparing to unpack .../libcurl3-gnutls_7.88.1-10+deb12u8_arm64.deb ...
2.923 Unpacking libcurl3-gnutls:arm64 (7.88.1-10+deb12u8) over (7.88.1-10+deb12u1) ...
2.960 Setting up libcurl3-gnutls:arm64 (7.88.1-10+deb12u8) ...
2.963 Setting up libcurl4:arm64 (7.88.1-10+deb12u8) ...
2.967 Setting up curl (7.88.1-10+deb12u8) ...
2.971 Setting up libcurl4-openssl-dev:arm64 (7.88.1-10+deb12u8) ...
2.974 Processing triggers for libc-bin (2.36-9+deb12u1) ...
3.000 mv: cannot stat 'app/kaggle.json': No such file or directory
------
failed to solve: process "/bin/sh -c apt-get update && apt-get install -y     curl     unzip     && rm -rf /var/lib/apt/lists/*     && mkdir app/data     && mkdir -p /root/.kaggle/     && mv app/kaggle.json /root/.kaggle/kaggle.json     && chmod 600 /root/.kaggle/kaggle.json     && pip3 install -r app/requirements.txt     && kaggle datasets download -d ${DATA}     && unzip *.zip -d ./app/data     && rm -rf *.zip     && cd app" did not complete successfully: exit code: 1
```

**Comment by H1mansh0** [comment]
To make starter bash script work, you need to have kaggle.json file (API key from Kaggle)

File: _dbt/models/dimension/dim_raw__date.sql
Comment (maksymloktionov) [inline _dbt/models/dimension/dim_raw__date.sql:1]: Please try to use this package here: https://hub.getdbt.com/calogica/dbt_date/latest/

File: _dbt/parser.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        except Exception as e:
+            logging.exception(f"Error: {e}")
+
+    def _start_connections(self) -> sqlalchemy.Engine:
+        try:
+            engine = sqlalchemy.create_engine(
+                f"postgresql+psycopg2://{self.user}:{self.paswrd}@{self.host}:{self.port}/{self.dbname}"
+            )
+
+            print(f"postgresql+psycopg2://{self.user}:{self.paswrd}@{self.host}:{self.port}/{self.dbname}")
```
Comment (maksymloktionov) [inline _dbt/parser.py:52]: Let's not print secure variables, even for local development

**Review by maksymloktionov — APPROVED** [review]

**Pull request merged**